source/triggers/clauses/clause_OnBlockMissing.js

```diff
@@ -0,0 +1,76 @@
+// Copyright 2014 Lockheed Martin Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may 
+// not use this file except in compliance with the License. You may obtain 
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS, 
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+this.onGenerated = function( params, generator, payload ) {
+    if ( !params || ( params.length != 2 ) ) {
+        this.logger.errorx( "onGenerated", 
+                            "This clause requires two arguments: " +
+                            "The name of the rover who's blockly window will be running " + 
+                            "and the type of block we want present in our main workspace.");
+        return false;
+    }
+
+    if ( !this.initOnEvent( params, generator, payload ) ) {
+        return false;
+    }
+
+    this.blockType = params[ 1 ];    
+
+    // Setup the callbacks that should trigger us
+    var object = this.findInScene( params[ 0 ] );
+
+    if ( object.blocklyStarted ) {
+        object.blocklyStarted = this.events.add( function() { this.onBlockMissing(); }, this );
+    } else {
+        this.logger.warnx( "onGenerated", "blocklyStarted event not " +
+                            "found for '" + object.name + "'." );
+    }
+
+    // Setup the callbacks that should prevent us from triggering
+    var object = this.findInScene( params[ 0 ] );
+
+    if ( object.blocklyStopped ) {
+        object.blocklyStopped = this.events.add( function() { this.reset(); }, this );
+    } else {
+        this.logger.warnx( "onGenerated", "blocklyStopped event not " +
+                            "found for '" + object.name + "'." );
+    }
+
+    if ( object.blocklyErrored ) {
+        object.blocklyErrored = this.events.add( function() { this.reset(); }, this );
+    } else {
+        this.logger.warnx( "onGenerated", "blocklyErrored event not " +
+                            "found for '" + object.name + "'." );
+    }
+
+    return true;
+}
+
+this.onBlockMissing = function() {
+
+    var workspace = Blockly.getMainWorkspace();
+    var presentBlocks = workspace.getAllBlocks();
+    var present = false;
+    for (var i = 0; i < presentBlocks.length; i++) {
```

Comment:
It's a small thing, but the VWF folks are sticklers for their particular coding style - which, among other things, means putting spaces inside of parentheses.